src/gollum/sonora.py

```diff
@@ -123,17 +123,111 @@ def __init__(
         else:
             super().__init__(*args, **kwargs)
 
+class Sonora2021Spectrum(PrecomputedSpectrum):
+    r"""
+    A container for a single Sonora precomputed synthetic spectrum of a brown dwarfs or free-floating
+    Gas Giant planet.
+
+    Args:
+        Teff (int): The Teff label of the Sonora model to read in.  Must be on the Sonora grid.
+        logg (float): The logg label of the Sonora model to read in.  Must be on the Sonora grid.
+        path (str): The path to your local Sonora grid library.  You must have the Sonora grid downloaded locally.  Default: "~/libraries/raw/Sonora/"
+        wl_lo (float): the bluest wavelength of the models to keep (Angstroms)
+        wl_hi (float): the reddest wavelength of the models to keep (Angstroms)
+        """
+
+    def __init__(
+        self, *args, teff=None, logg=None, met = None, path=None, wl_lo=8038, wl_hi=12849, **kwargs
+    ):
+
+        teff_points = np.hstack(
+            (
+                np.arange(500, 600, 25),
+                np.arange(600, 1000, 50),
+                np.arange(1000, 2401, 100),
+            )
+        )
+        logg_points = np.arange(4.0, 5.51, 0.25)
+
+        # Map logg (cgs) to the gravity labels used in file names
+        logg_par_dict = {
+            4.0: "100",
+            4.25: "178",
+            4.5: "316",
+            4.75: "562",
+            5.0: "1000",
+            5.25: "1780",
+            5.5: "3160",
+        }
+
+        met_points = [-0.5, 0, 0.5]
+
+        if path is None:
+            path = "~/libraries/raw/SonoraBobcat2021/"
+
+        if (teff is not None) & (logg is not None) & (met is not None):
+            base_path = os.path.expanduser(path)
+            assert os.path.exists(
+                base_path
+            ), "You must specify the path to local Sonora models: {}".format(base_path)
+
+            print("successfully found the path to Sonora models: {}".format(base_path))
+
+            assert teff in teff_points, "Teff must be on the grid points"
+            assert logg in logg_points, "logg must be on the grid points"
+            assert met in met_points, "Fe/H must be a valid point"
+
+            if met < 0:
+                base_name = "sp_t{0:0>.0f}g{1:}nc_m{0:0.01f}".format(
+                    float(teff), logg_par_dict[logg], float(met)
+                    )
+            else:
+                base_name = "sp_t{0:0>.0f}g{1:}nc_m{0:+0.0f}".format(
+                    float(teff), logg_par_dict[logg], float(met)
+                    )
+            fn = base_path + "/" + base_name
```

Comment:
This hardcoded path separator will likely break for Windows, but it will work for Mac and Linux.